### PR TITLE
core/vm: clear linter warnings

### PR DESCRIPTION
**Pull request by gballet**
As i started working on the intPool pool, I got welcomed with a whole bunch of linter errors. This PR quells them.

### CODE REVIEW

File: core/vm/logger.go [hunk 1]
```diff
@@ -29,8 +29,10 @@ import (
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
+// Storage represents a contract's storage
```
Comment (karalabe) [inline core/vm/logger.go]: period

File: core/vm/logger.go [hunk 2]
```diff
@@ -29,8 +29,10 @@ import (
...
 type Storage map[common.Hash]common.Hash
 
+// Copy duplicates the current storage
```
Comment (karalabe) [inline core/vm/logger.go]: rather emphasize that it's a deep copy. also period

File: core/vm/logger.go [hunk 3]
```diff
@@ -76,10 +78,12 @@ type structLogMarshaling struct {
 	ErrorString string `json:"error"`  // adds call to ErrorString() in MarshalJSON
 }
 
+// OpName formats the
```
Comment (karalabe) [inline core/vm/logger.go]: the end is missing, period

File: core/vm/logger.go [hunk 4]
```diff
@@ -76,10 +78,12 @@ type structLogMarshaling struct {
...
 func (s *StructLog) OpName() string {
 	return s.Op.String()
 }
 
+// ErrorString formats the log's error as a string
```
Comment (karalabe) [inline core/vm/logger.go]: period

File: core/vm/logger.go [hunk 5]
```diff
@@ -124,6 +128,7 @@ func NewStructLogger(cfg *LogConfig) *StructLogger {
 	return logger
 }
 
+// CaptureStart logs the start of a contract
```
Comment (karalabe) [inline core/vm/logger.go]: // CaptureStart implements the Tracer interface to initialize the tracing operation.

File: core/vm/logger.go [hunk 6]
```diff
@@ -178,10 +183,12 @@ func (l *StructLogger) CaptureState(env *EVM, pc uint64, op OpCode, gas, cost ui
 	return nil
 }
 
+// CaptureFault logs a fault in the execution
```
Comment (karalabe) [inline core/vm/logger.go]: // CaptureFault implements the Tracer interface to trace an execution fault
// while running an opcode.

File: core/vm/logger.go [hunk 7]
```diff
@@ -178,10 +183,12 @@ func (l *StructLogger) CaptureState(env *EVM, pc uint64, op OpCode, gas, cost ui
...
 func (l *StructLogger) CaptureFault(env *EVM, pc uint64, op OpCode, gas, cost uint64, memory *Memory, stack *Stack, contract *Contract, depth int, err error) error {
 	return nil
 }
 
+// CaptureEnd logs the end of a contract
```
Comment (karalabe) [inline core/vm/logger.go]: // CaptureEnd is called after the call finishes to finalize the tracing.

File: core/vm/memory.go [hunk 8]
```diff
@@ -29,6 +29,7 @@ type Memory struct {
 	lastGasCost uint64
 }
 
+// NewMemory returns a new memory memory model
```
Comment (karalabe) [inline core/vm/memory.go]: period

File: core/vm/memory.go [hunk 9]
```diff
@@ -107,6 +108,7 @@ func (m *Memory) Data() []byte {
 	return m.store
 }
 
+// Print shows the content of the memory
```
Comment (karalabe) [inline core/vm/memory.go]: `shows` - > `dumps`. period

File: core/vm/noop.go [hunk 10]
```diff
@@ -23,48 +23,105 @@ import (
...
+// Snapshot dummy method
+func (NoopStateDB) Snapshot() int { return 0 }
+
+// AddLog dummy method
+func (NoopStateDB) AddLog(*types.Log) {}
+
+// AddPreimage dummy method
+func (NoopStateDB) AddPreimage(common.Hash, []byte) {}
+
+// ForEachStorage dummy method
```
Comment (karalabe) [inline core/vm/noop.go]: Please revert these changes. These don't add any meaningful information to the code. On the other hand, the many lines bloat up the code and make it harder to see that this is a noop mock construct.

File: core/vm/opcodes.go [hunk 11]
```diff
@@ -23,6 +23,7 @@ import (
 // OpCode is an EVM opcode
 type OpCode byte
 
+// IsPush specifies if an opcode is a PUSH opcode
```
Comment (karalabe) [inline core/vm/opcodes.go]: period

File: core/vm/opcodes.go [hunk 12]
```diff
@@ -31,12 +32,13 @@ func (op OpCode) IsPush() bool {
 	return false
 }
 
+// IsStaticJump specifies if an opcode is JUMP
```
Comment (karalabe) [inline core/vm/opcodes.go]: period

File: core/vm/stack.go [hunk 13]
```diff
@@ -32,6 +32,7 @@ func newstack() *Stack {
 	return &Stack{data: make([]*big.Int, 0, 1024)}
 }
 
+// Data returns the underlying big.Int array
```
Comment (karalabe) [inline core/vm/stack.go]: period

File: core/vm/stack.go [hunk 14]
```diff
@@ -80,6 +81,7 @@ func (st *Stack) require(n int) error {
 	return nil
 }
 
+// Print displays the content of the stack
```
Comment (karalabe) [inline core/vm/stack.go]: period

Comment (karalabe) [reply core/vm/stack.go]: `displays` -> `dumps`

**Comment by karalabe** [comment]
I still want that revert :D The old code looked like a nice table where you could see what the noop methods returns. In the new code, the details go missing among the many boilerplate.

**Comment by gballet** [comment]
I'm aware, I am taking a few hours to see if I can come with something that gets rid of these warnings.
